### DISCUSSION TITLE
Is it possible to keep a Menu open when clicking outside?

### DISCUSSION THREAD

**Discussion by kboul** [microsoft/fluentui#28183] [Q&A] [answered]
I am using `"@fluentui/react-components": "^9.20.3"` and I would like to ask if it is possible to keep a Menu component open even if I click outside of the Menu. I have a table where upon clicking a table row using a checkbox I would like a specific Menu to be kept open because I display row information inside. My Menu code is as simple as that 

 ```
 <Menu>
    <MenuTrigger disableButtonEnhancement>
      <Button>Toggle menu</Button>
    </MenuTrigger>

    <MenuPopover>
      <MenuList>
        <MenuItem>New </MenuItem>
        <MenuItem>New Window</MenuItem>
        <MenuItem disabled>Open File</MenuItem>
        <MenuItem>Open Folder</MenuItem>
      </MenuList>
    </MenuPopover>
  </Menu>
```

I also tried using the Dialog of type non-modal but it is always centered on the screen. I would like to be displayed with a reference to its trigger point.

I looked on the [docs](https://react.fluentui.dev/?path=/docs/components-menu-menu--default) but there not seems to be a relevant prop for what I am looking for. Any recommendations are welcome.

**Comment by layershifter** [comment, marked as answer]
TL;DR make `Menu` controlled and check event's data:

```tsx
import {
  Button,
  Menu,
  MenuTrigger,
  MenuList,
  MenuItem,
  MenuPopover,
  MenuProps
} from "@fluentui/react-components";
import * as React from "react";

export const ControllingOpenAndClose = () => {
  const [open, setOpen] = React.useState(false);
  const onOpenChange: MenuProps["onOpenChange"] = (e, data) => {
    if (data.type === "clickOutside") {
      return;
    }

    setOpen(data.open);
  };

  return (
    <div>
      <Menu open={open} onOpenChange={onOpenChange}>
        <MenuTrigger disableButtonEnhancement>
          <Button>Toggle menu</Button>
        </MenuTrigger>

        <MenuPopover>
          <MenuList>
            <MenuItem>New </MenuItem>
            <MenuItem>New Window</MenuItem>
            <MenuItem disabled>Open File</MenuItem>
            <MenuItem>Open Folder</MenuItem>
          </MenuList>
        </MenuPopover>
      </Menu>
    </div>
  );
};
```

https://codesandbox.io/s/elastic-hill-xvj7st?file=/example.tsx

**Reply by kboul** [reply]
That is great. Thank you!
